File: System/Protocol/Allors.Protocol.Direct/Api/Pull/PullRequest.cs
// <copyright file="PullRequest.cs" company="Allors bvba">
// Copyright (c) Allors bvba. All rights reserved.
// Licensed under the LGPL license. See LICENSE file in the project root for full license information.
// </copyright>

namespace Allors.Protocol.Direct.Api.Pull
{
    using Workspace.Data;

    public class PullRequest
    {
        public Pull[] Pulls { get; set; }
    }
}

File: Core/Workspace/Typescript/libs/protocol/core/src/lib/pull/PullRequest.ts
import { Procedure, Pull } from '@allors/data/core';

export class PullRequest {
  public pulls?: Pull[];

  public procedure?: Procedure;

  constructor(fields?: Partial<PullRequest>) {
    Object.assign(this, fields);
  }

  public toJSON() {
    return {
      p: this.pulls,
      s: this.procedure,
    };
  }
}
